test/utmost/unwrap.js

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const Utmost = artifacts.require('UtmostTest');
+
+const BN = require('bn.js');
+
+const Utils = require('./../test_lib/utils');
+const EventDecoder = require('./../test_lib/event_decoder.js');
+const { AccountProvider } = require('./../test_lib/utils');
+
+contract('Utmost.unwrap()', (accounts) => {
+  let utmost;
+  let initialSupply;
+  let caller;
+  let wrappedAmount;
+  let amountToUnwrap;
+  let coconsensus;
+  const accountProvider = new AccountProvider(accounts);
+
+  beforeEach(async () => {
+    coconsensus = accountProvider.get();
+    initialSupply = new BN('1000000');
+    utmost = await Utmost.new(coconsensus, initialSupply);
+    await utmost.setup({ from: coconsensus });
+    caller = accountProvider.get();
+    amountToUnwrap = new BN('20');
+    // Mints ERC20 Utmost balance for caller
+    wrappedAmount = new BN('100');
+    await utmost.wrap({ from: caller, value: wrappedAmount });
+  });
+
+  it('should unwrap successfully with correct parameters', async () => {
+    const initialContractBaseCoinBalance = await Utils.getBalance(utmost.address);
+    const initialCallerBaseCoinBalance = await Utils.getBalance(caller);
+
+    const result = await utmost.unwrap.call(amountToUnwrap, {
+      from: caller,
+    });
+    assert.strictEqual(result, true, 'The contract should return true.');
+
+    const tx = await utmost.unwrap(amountToUnwrap, { from: caller });
+    const gasUsed = new BN(tx.receipt.gasUsed);
+
+    const callerERC20TokenBalance = await utmost.balanceOf.call(
+      caller,
+    );
+    assert.strictEqual(
+      callerERC20TokenBalance.eqn(80),
```

Comment:
Instead of hardcoded value please use `wrappedAmount .sub(amountToUnwrap)`

Reply:
Updated!